src/pages/index.js

```diff
@@ -0,0 +1,381 @@
+    })
+    .catch((error) => {
+      console.error("Failed to edit user info:", error);
+    })
+    .finally(() => {
+      setButtonText(editSubmitBtn, false);
+    });
+}
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(cardSubmitBtn, true);
+  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
+  api
+    .addCard(inputValues)
+    .then((data) => {
+      const cardElement = getCardElement(data);
+      cardsList.prepend(cardElement);
+      closeModal(cardModal);
+      cardForm.reset(); // Clear inputs after successful submission
+      disableButton(cardSubmitBtn, validationConfig.inactiveButtonClass); // Disable button after successful submission
+    })
+    .catch((error) => {
+      console.error("Failed to add card:", error);
+    })
+    .finally(() => {
+      setButtonText(cardSubmitBtn, false);
+    });
+}
+
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  setButtonText(avatarSubmitBtn, true);
+  api
+    .editUserAvatar(avatarInput.value)
+    .then((data) => {
+      profileAvatar.src = data.avatar;
+      closeModal(avatarModal);
+      disableButton(avatarSubmitBtn, validationConfig.inactiveButtonClass); // Disable button after successful submission
+    })
+    .catch((error) => {
+      console.error("Failed to edit user avatar:", error);
+    })
+    .finally(() => {
+      setButtonText(avatarSubmitBtn, false);
+    });
+}
+
+// function disableButton(button, inactiveButtonClass) {
+//   button.classList.add(inactiveButtonClass);
+//   button.disabled = true;
+// }
+
+// function hideInputError(formElement, inputElement, validationConfig) {
+//   const errorElement = formElement.querySelector(`.${inputElement.id}-error`);
+//   if (errorElement) {
+//     inputElement.classList.remove(validationConfig.inputErrorClass);
+//     errorElement.classList.remove(validationConfig.errorClass);
+//     errorElement.textContent = "";
+//   }
+// }
+
+// function toggleButtonState(inputList, buttonElement, validationConfig) {
+//   if (hasInvalidInput(inputList)) {
+//     buttonElement.classList.add(validationConfig.inactiveButtonClass);
+//     buttonElement.disabled = true;
+//   } else {
+//     buttonElement.classList.remove(validationConfig.inactiveButtonClass);
+//     buttonElement.disabled = false;
+//   }
+// }
+
+// function hasInvalidInput(inputList) {
+//   return inputList.some((inputElement) => {
+//     return !inputElement.validity.valid;
+//   });
+// }
+
+previewModalCloseButton.addEventListener("click", () => {
+  closeModal(previewModal);
+});
+
+cardModalCloseButton.addEventListener("click", () => {
+  closeModal(cardModal);
+});
+
+// Open avatar modal
+avatarEditButton.addEventListener("click", () => {
+  openModal(avatarModal);
+  avatarForm.reset();
+  disableButton(avatarSubmitBtn, validationConfig.inactiveButtonClass);
+});
+avatarForm.addEventListener("submit", handleAvatarSubmit);
```

Comment:
Please, delete `avatarForm.addEventListener("submit", handleAvatarSubmit);` because you have the avatar submission handler in line 353